src/components/TopBar/TopBar.jsx

```diff
@@ -1,2 +1,115 @@
-import React from "react";
-export const TopBar = () => <>TopBar</>;
+import { React, useState } from "react";
+import {
+  StyledTopBarWrapper,
+  StyledTitle,
+  StyledToolsWrapper,
+  StyledSvg,
+  StyledLineSvg,
+  StyledSvgWrapper,
+  StyledName,
+  StyledImgBorder,
+  StyledImg,
+  StyledInput,
+} from "./style";
+import { useLocation } from "react-router-dom";
+import profilePhoto from "../../images/profile-photo.png";
+
+export const TopBar = () => {
+  const location = useLocation();
+  const [isInputDisplayed, setIsInputDisplayed] = useState(false);
+
+  const handleSearchClick = () => {
+    setIsInputDisplayed(!isInputDisplayed);
+  };
+
+  const getTitle = () => {
+    switch (location.pathname) {
+      case "/dashboard/Overview":
+        return "Overview";
+      case "/dashboard/Tickets":
+        return "Tickets";
+      case "/dashboard/Ideas":
+        return "Ideas";
+      case "/dashboard/Contacts":
+        return "Contacts";
+      case "/dashboard/Agents":
+        return "Agents";
+      case "/dashboard/Articles":
+        return "Articles";
+      case "/dashboard/Settings":
+        return "Settings";
+      case "/dashboard/Subscription":
+        return "Subscription";
+      default:
+        return "Error";
+    }
+  };
+  return (
+    <StyledTopBarWrapper>
+      <StyledTitle>{getTitle()}</StyledTitle>
+      <StyledToolsWrapper>
+        <StyledSvgWrapper>
+          <StyledInput isInputDisplayed={isInputDisplayed} />
+          <StyledSvg
```

Comment:
Please move svg itself to a different file (probably inside `assets` folder) and check with create-react-app docs how to import it as a component